internal/orchestration/controller.go

```diff
@@ -342,14 +343,31 @@ func (e *Experiment) stopReplicas() error {
 		for id, hash := range res.GetHashes() {
 			hashes[id] = hash
 		}
+		for id, cnt := range res.GetCounts() {
+			counts[id] = cnt
+		}
+	}
+	for id1, cnt1 := range counts {
+		for id2, cnt2 := range counts {
+			if id2 > id1 && cnt2 == cnt1 {
+				if !bytes.Equal(hashes[id1], hashes[id2]) {
+					return fmt.Errorf("hash mismatch for same count")
+				}
+			}
+		}
 	}
-	var cmp []byte
-	for _, hash := range hashes {
-		if cmp == nil {
-			cmp = hash
+	var count uint32
+	first := true
```

Comment:
Not sure what happened here, but in my browser, it looks like there is a newline after the `=` sign. But I don't see it in VSCode. Maybe it will go away with a new commit touching this file...